web-app/js/portal/filter/combiner/GaWcsParametersBuilder.js

```diff
@@ -0,0 +1,202 @@
+Ext.namespace('Portal.filter.combiner');
+
+Portal.filter.combiner.GaWcsParametersBuilder = Ext.extend(Portal.filter.combiner.BaseFilterCombiner, {
+
+    buildWCSParameters: function(layerName) {
+
+        var geomFilter = this.filters.map(function(filter) {
+
+            if (filter.constructor == Portal.filter.GeometryFilter) {
+                return filter;
+            }
+        });
+
+        return this.buildParameters(geomFilter[0], layerName);
+    },
+
+    buildParameters: function(geomFilter, layerName) {
+
+        var layerConfig = this.getLayerConfig(layerName);
+        layerConfig.selectedBBOX = (geomFilter.value != undefined) ? geomFilter.value.getBounds().toBBOX() : geomFilter.map.getExtent().toBBOX();
+        layerConfig.size = this.getGridEnvelope(layerConfig);
+        return layerConfig;
+    },
+
+    getLayerConfig: function(layerName) {
+
+        var clipConfig = this.getClipConfig();
+        var selected = {};
+
+        Ext.each(clipConfig.datasets, function(dataset) {
+             if (dataset.layers ==  layerName) {
+                 Ext.apply(selected, dataset);
+             }
+        }, this);
+
+        return selected;
+
+    },
+
+    getGridEnvelope: function (wcsParameters) {
+
+        var maxBounds = new OpenLayers.Bounds(wcsParameters.maxBoundingBox);
+        var newBounds = new OpenLayers.Bounds(wcsParameters.selectedBBOX.split(','));
+        var widthPercentage = newBounds.getWidth() / maxBounds.getWidth();
+        var heightPercentage = newBounds.getHeight() / maxBounds.getHeight();
+        return {
+            width: ((wcsParameters.maxGridEnvelope[0] * widthPercentage) * wcsParameters.scaleFactor).toFixed(0),
+            height: ((wcsParameters.maxGridEnvelope[1] * heightPercentage) * wcsParameters.scaleFactor).toFixed(0)
+        };
+    },
+
+    getClipConfig: function() {
```

Comment:
NB: for prototype only.  Need to use GA service that uses these internally when available or moved to external config.